distributed/http/worker/tests/test_worker_http.py

```diff
@@ -1,32 +1,116 @@
 from __future__ import annotations
 
+import asyncio
 import json
 
 import pytest
 from tornado.httpclient import AsyncHTTPClient
 
+from distributed import Event
 from distributed.utils_test import gen_cluster
 
 
-@gen_cluster(client=True)
-async def test_prometheus(c, s, a, b):
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 1)])
+async def test_prometheus(c, s, a):
     pytest.importorskip("prometheus_client")
     from prometheus_client.parser import text_string_to_metric_families
 
     http_client = AsyncHTTPClient()
 
+    async def fetch_metrics():
```

Comment:
Seems like `fetch_metrics` could be factored out to a `utils_test` function?

Reply:
Done